### ISSUE TITLE
Classic changed Enchanting Recipes

### ISSUE THREAD

**Issue by Hacki95** [azerothcore/mod-progression-system#61]
Arbitration made me aware of an issue we will face in the 50-60 bracket.

Most of the classic enchantments for caster classes got changed with wotlk, while they required items like "nexus crystals" before (disenchanted from classic items) they now require "prismatic shards" (disenchanted from TBC items only).

This is true for most other ingredients as well (Elemental items, Dust etc).
Is it possible to change the requirements in a recipe? Not without Core edits right?
What are our possibilities here?
1) Try to change back the recipes to classic

From my understanding since these spells are saved in the client there is no simple way to just overwrite these with sql values in the DB. If I am wrong please correct me here but I think this would require a custom patch.
2) Make an exchange vendor

We would need to find out if there is an almost 1:1 way to exchange the materials to the classic values. Like @Yehonal said this might be possible to do without core edits but I am not sure.
3) Make an exchange gossip NPC

If the option of using a normal vendor doesn't work we could script a simple exchange gossip NPC. Same principle as for option 2, we would need to research first if the items can be exchanged 1:1
So what should we do??

We need to find someone who has the time to compare the classic recipes with the changed ones to find out how we could exchange the items accordingly because from my understanding changing the recipes would mean too much work and a custom patch.

**Comment by Shineslippers** [comment]
### Enchanting

There is only recipes what changed mats to TBC one. Existing changed mats cost recipes, but it's not important.

Recipe | Old | Current
 -- | -- | -- 
Cloak - Dodge [25086](https://wowgaming.altervista.org/aowow/?spell=25086)  | 3xNC 8xLBS 8xGuardian Stone| 3xSPS 3xGPE 8xPrimal Earth
Cloak - Greater Fire Resistance [25081](https://wowgaming.altervista.org/aowow/?spell=25081) | 3xNC 8xLBS 4xEssence of Fire | 3xSPS 3xGPE 4xEssence of Fire
Cloak - Greater Nature Resistance [25082](https://wowgaming.altervista.org/aowow/?spell=25082) | 2xNC 8xLBS 4xLiving Essence | 2xSPS 3xGPE 4xLiving Essence
Cloak - Stealth [25083](https://wowgaming.altervista.org/aowow/?spell=25083) | 3xNC 8xLBS 2xBlack Lotus | 3xSPS 3xGPE 2xFel Lotus
Cloak - Subtlety [25084](https://wowgaming.altervista.org/aowow/?spell=25084) | 4xNC 6xLBS 2xBlack Diamond | 4xSPS 2xGPE 8xPrimal Shadow
Gloves - Fire Power [25078](https://wowgaming.altervista.org/aowow/?spell=25078) | 2xNC 10xLBS 4xEssence of Fire | 2xSPS 10xLBS 4xEssence of Fire
Gloves - Frost Power [25074](https://wowgaming.altervista.org/aowow/?spell=25074) |  3xNC 10xLBS 4xEssence of Water | 3xSPS 10xLBS 4xEssence of Water
Gloves - Healing Power [25079](https://wowgaming.altervista.org/aowow/?spell=25079) | 3xNC 8xLBS 1xRighteous Orb | 3xSPS 8xLBS 1xRighteous Orb
Gloves - Shadow Power [25073](https://wowgaming.altervista.org/aowow/?spell=25073) |  3xNC 10xLBS 6xEssence of Undeath | 3xSPS 10xLBS 6xEssence of Undeath
Gloves - Superior Agility [25080](https://wowgaming.altervista.org/aowow/?spell=25080) | 3xNC 8xLBS 4xEssence of Air | 3xSPS 3xGPE 2xPrimal Air
 Gloves - Threat [25072](https://wowgaming.altervista.org/aowow/?spell=25072) | 4xNC 6xLBS 8xLarval Acid | 4xSPS 2xGPE 8xPrimal Life

Possible ratio:
- 1x[Nexus Crystal](https://wowgaming.altervista.org/aowow/?item=20725) to 1x[Small Prismatic Shard](https://wowgaming.altervista.org/aowow/?item=22448)
- 3x[Large Brilliant Shard](https://wowgaming.altervista.org/aowow/?item=14344) to 1x[Greater Planar Essence](https://wowgaming.altervista.org/aowow/?item=22446)
- 1x[Guardian Stone](https://wowgaming.altervista.org/aowow/?item=12809) to 1x[Primal Earth](https://wowgaming.altervista.org/aowow/?item=22452)
- 1x[Black Lotus](https://wowgaming.altervista.org/aowow/?item=13468) to 1x[Fel Lotus](https://wowgaming.altervista.org/aowow/?item=22794)
- ?x[Black Diamond](https://wowgaming.altervista.org/aowow/?item=11754) to ?x[Primal Shadow](https://wowgaming.altervista.org/aowow/?item=22456). Primal Shadow not so trashy to exchange 1x4, but this debatable
- 2x[Essence of Air](https://wowgaming.altervista.org/aowow/?item=7082) to 1x[Primal Air](https://wowgaming.altervista.org/aowow/?item=22451)
- 1x[Larval Acid](https://wowgaming.altervista.org/aowow/?item=18512) to 1x[Primal Life](https://wowgaming.altervista.org/aowow/?item=21886)

**Comment by UltraNix** [comment]
There is no option to change reagents without clientside modifications.
Also - an exchange vendoris not a good choice - people can abuse and farm mats, exchange for TBC mats and then wait till TBC release. It would kill AH/economy.

**Comment by 55Honey** [comment]
> There is no option to change reagents without clientside modifications. Also - an exchange vendoris not a good choice - people can abuse and farm mats, exchange for TBC mats and then wait till TBC release. It would kill AH/economy.

Good point, but we can just run a query and convert them all back during the update from vanilla to BC.

https://github.com/55Honey/Acore_ExchangeNpc
Here is the first draft of the NPC. Values are only for testing. There's a NPC Insert query missing in it for the creature_template table.

**Comment by UltraNix** [comment]
`Good point, but we can just run a query and convert them all back during the update from vanilla to BC.`
Sounds like a good cause of drama :)

**Comment by 55Honey** [comment]
We need to make it clear that the Mats would be converted back from the start. That's the best we can do.

**Comment by UltraNix** [comment]
And what about money? 
`Of course, people just can farm these mats, exchange and sell to vendor to make money - it's another example.`
I think it's not a best idea.

**Comment by 55Honey** [comment]
That sounds like we need to lower the vendor value of the BC items / set it to 0 temporarily. Good point actually!

**Comment by UltraNix** [comment]
@55Honey We want a new exchange vendor for that?

**Comment by 55Honey** [comment]
We can add all of these mats to the one that exists. We'll test it asap and adjust the values/mats.

**Comment by UltraNix** [comment]
"The one that exists" - which is...?

**Comment by 55Honey** [comment]
This one :)

> > There is no option to change reagents without clientside modifications. Also - an exchange vendoris not a good choice - people can abuse and farm mats, exchange for TBC mats and then wait till TBC release. It would kill AH/economy.
> 
> Good point, but we can just run a query and convert them all back during the update from vanilla to BC.
> 
> https://github.com/55Honey/Acore_ExchangeNpc Here is the first draft of the NPC. Values are only for testing. There's a NPC Insert query missing in it for the creature_template table.

**Comment by UltraNix** [comment]
@55Honey I will move it to this module.

**Comment by UltraNix** [comment]
@55Honey So you are working on it?

**Comment by 55Honey** [comment]
Yes. We did some testing today. We're going to add a confirmation gossip and then it should be good to go.

**Comment by 55Honey** [comment]
Thanks! I didn't think of that tbh :-)

**Comment by 55Honey** [comment]
Someone, please test the PR above, and check the ratios so we can close this.

**Comment by Helias** [comment]
Tested, hope we can use it soon 🚀

**Comment by 55Honey** [comment]
@Hacki95 please check the ratios. If fine, we can merge and close this.
![image](https://user-images.githubusercontent.com/71938210/144672470-b6025778-f96e-4846-af23-214568df0b50.png)

**Comment by Shineslippers** [comment]
@55Honey Can this NCP change several item to one? I mean if for full set of mats gives enchant `Scroll` instead of mats. Idk is this any better or not, but in this case it will not affect other proffesions

**Comment by 55Honey** [comment]
Sure, we could do that. But it would remove the meaning of the actual recipes from the game. People should be required to farm them in the raids.

I don't see an effect on other professions, other than the prices tbh.

**Comment by Hacki95** [comment]
We should also keep in mind, before TBC is released we should convert ALL tbc materials back to classic ones if people stockpiled them. That way nobody can prefarm TBC mats.

**Comment by 55Honey** [comment]
@Hacki95 The NPC mentions that in the confirmation dialog. The query would be manual and needs to be added to the progression repo as a one-time execution.
![image](https://user-images.githubusercontent.com/71938210/144748196-76744046-ba1e-4a25-9da9-ed9af333de80.png)

**Comment by Hacki95** [comment]
neat
